src/SignalR/clients/csharp/Client/src/HttpConnectionFactory.cs

```diff
@@ -16,16 +18,23 @@ namespace Microsoft.AspNetCore.SignalR.Client
     /// </summary>
     public class HttpConnectionFactory : IConnectionFactory
     {
+        private readonly TransferFormat _transferFormat; 
         private readonly HttpConnectionOptions _httpConnectionOptions;
         private readonly ILoggerFactory _loggerFactory;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="HttpConnectionFactory"/> class.
         /// </summary>
+        /// <param name="hubProtocol">The <see cref="IHubProtocol"/> that will use the connections.</param>
         /// <param name="options">The connection options.</param>
         /// <param name="loggerFactory">The logger factory.</param>
-        public HttpConnectionFactory(IOptions<HttpConnectionOptions> options, ILoggerFactory loggerFactory)
+        public HttpConnectionFactory(IHubProtocol hubProtocol, IOptions<HttpConnectionOptions> options, ILoggerFactory loggerFactory)
         {
+            if (hubProtocol == null)
```

Comment:
I would, but throw expressions don't look as nice once you start trying to pull out a specific property like IHubProtocol.TransferFormat and store that in the field instead of the entire argument. You end up needing a ternary expression or something. I assume this is the same reason we didn't use a throw expression for options and loggerFactory.